File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        int[] res=new int[2];
        int n=nums.length;
        Map<Integer,Integer>mp=new HashMap<>();
    
        for(int i=0;i<n;i++)
        {
            if(mp.containsKey(nums[i]))
            {
                res[0]=i;
                res[1]=mp.get(nums[i]);
            }
            else
            {
                mp.put(target-nums[i],i);
            }
        }
        return res;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n=nums.size(),diff;
        map<int,int>mp;
        vector<int>ans;
        
        for(int i=0;i<n;i++) {
            mp[nums[i]]=i+1;
        }
        
        for(int i=0;i<n;i++) {
            diff=target-nums[i];
            if((mp[diff])&&(mp[diff]!=i+1)) {
                ans.push_back(i);
                ans.push_back(mp[diff]-1);
                break;
            }
        }
        
        
        return ans;
    }
};